### DISCUSSION TITLE
Log Analytics shared workspace costs

### DISCUSSION THREAD

**Discussion by PaulinaBadura2** [microsoft/finops-toolkit#1084] [Q&A] [answered]
I'm exploring ways to manage cost recharging for shared Azure Log Analytics workspaces, where costs need to be allocated across multiple recharge entities (e.g., different departments or business units). Currently, it's challenging to accurately allocate Log Analytics costs by entity, and querying data tables directly has been hard. I’m considering whether the Azure FinOps Toolkit, or any other automated tools or scripts, could help streamline this process. Has anyone successfully used this toolkit (or similar solutions) to handle cost allocation in shared workspaces? Any advice on approaches or automation tips would be greatly appreciated!

**Comment by helderpinto** [comment, marked as answer]
@PaulinaBadura2, does [this article](https://techcommunity.microsoft.com/t5/core-infrastructure-and-security/how-to-allocate-azure-monitor-logs-ingestion-costs-by-resource/ba-p/3813322) help? It's not part of the FinOps toolkit yet. Your feedback would be helpful in improving the Workbook before bringing it in to the toolkit.

**Reply by PaulinaBadura2** [reply]
@helderpinto Thanks for your input. Could you please share the link again? I'm getting the 'Requested content does not exist' error while trying to access. Thanks.

**Reply by helderpinto** [reply]
You'll have to wait for a few hours more. The Tech Community platform is undergoing maintenance.

![image](https://github.com/user-attachments/assets/5a8cd6f5-d518-4b90-b8cb-5565c1c01eb0)

**Reply by helderpinto** [reply]
@PaulinaBadura2, the link is available now.

**Reply by estimpfle** [reply]
Hello.  Do you have a shared log analytics workspace that collects logging data for multiple entities or does each have it's own?
If it's a shared space, you can't get that out of the billing data.  You can look in Kusto logs, but it can be challenging to get out because logging can be very large, and it's not tagged.  If you cannot see it in the Microsoft Cost data, you will not be able to see it in FINOPS data set.  I suggest looking int he Microsoft data:  Filter on Log Analytics, Azure Monitor, and Sentinel. It's helpful to understand your AZ invoice that that level.  Take a look at which subscriptions or resource groups have billed costs for those meter categories.
